packages/api/endpoints/granules.js

```diff
@@ -110,36 +109,23 @@ const update = async (req, res) => {
     knex = await getKnexClient(),
   } = req.testContext || {};
   const body = req.body || {};
-  const { retries = 0, status, ...restOfBody } = body;
 
   let existingGranule;
-
   try {
-    existingGranule = await pRetry(
-      async () => await granuleModel.get({ granuleId: body.granuleId }),
-      {
-        retries,
-      }
-    );
+    existingGranule = await granuleModel.get({ granuleId: body.granuleId });
   } catch (error) {
-    if (error instanceof RecordDoesNotExist) {
-      return res.boom.notFound(`No granule found to update for ${body.granuleId}`);
+    if (body.status === 'queued') {
+      existingGranule = {};
```

Comment:
Is this logic here basically to handle the fact that we might be creating a new "queued" granule via this endpoint, so we might hit this catch but don't want to return an error from the endpoint?

Reply:
If so, then I think we're possibly confusing what a PUT endpoint should do. A PUT should create a new resource or update an existing one, meaning it shouldn't be relying on a check like this to see if a record already exists

https://developer.mozilla.org/en-US/docs/Web/HTTP/Methods/PUT

Now, implementing the endpoint that way may require some refactoring of our write logic, but it might be worth trying because it's ultimately a lot cleaner and easier to understand, and also compliant with how PUT endpoints generally work.

Reply:
That makes a lot of sense and was actually a really easy change. I kept in the try/catch so that if there is some other issue retrieving the record, it will still throw an error, but it'll never throw the 'record does not exist error'

Reply:
Just to be clear, I don't think we should need/have the `granuleModel.get` logic in this endpoint at all. A PUT endpoint should accept a full body representing a granule to be written, so there should be no need to look anything up from the database before doing the write.

Is there some reason that wouldn't be feasible here?

Reply:
Discussed offline and changing this would require larger refactoring to this endpoint logic which isn't essential for this PR, so resolving this comment